macros/parsers/parserPopUp.pl

```diff
@@ -256,23 +257,42 @@ sub MENU {
 #
 
 sub DropDown {
-	my ($self, $choices, $value, %options) = @_;
-	$options{placeholder} = '?' unless $options{placeholder};
-	unshift(@$choices, $options{placeholder});
-	my @order = map { ref($_) eq "ARRAY" ? @$_ : $_ } @$choices;
-	if ($value =~ m/^\d+$/ && $order[$value]) { $value++ }
-	return parser::PopUp->new($choices, $value, \%options);
+	my $self    = shift;
+	my $choices = shift;
+	my $value   = shift;
+	my %options = (
+		placeholder => '?',
+		@_
+	);
+	return parser::PopUp->new($choices, $value, %options);
 }
 
 #
 # TrueFalse() variant of PopUp()
 #
 
 sub DropDownTF {
-	my ($self, $value, %options) = @_;
-	my $sanitized_value = $value ? 'True' : 'False';
-	$sanitized_value = 'False' if (defined $value && $value =~ /^\s*(false|f)\s*$/i);
-	return parser::PopUp->DropDown([ 'True', 'False' ], $sanitized_value, %options, static => 0);
+	my $self    = shift;
+	my $value   = lc(main::maketext(shift));
+	my %options = (
+		static => 0,
+		@_
+	);
+	my $true         = main::maketext('True');
+	my $false        = main::maketext('False');
+	my %sanitization = (
+		lc($true)  => $true,
+		1          => $true,
+		lc($false) => $false,
+		0          => $false
+	);
+	if (lc(substr($true, 0, 1)) ne lc(substr($false, 0, 1))) {
+		$sanitization{ substr($true,  0, 1) } = $true;
+		$sanitization{ substr($false, 0, 1) } = $false;
```

Comment:
I think the substrings need to be lower-cased, here.  Other than that, I think it looks good.